news.xql

```diff
@@ -33,17 +33,23 @@ return
             <li>
                 <div class="date">
                 {
-                    if ($age < xdt:dayTimeDuration("PT1H")) then
-                        minutes-from-duration($age) || " minutes ago"
-                    else if ($age < xdt:dayTimeDuration("P1D")) then
-                        hours-from-duration($age) || " hours ago"
-                    else if ($age < xdt:dayTimeDuration("P14D")) then
-                        days-from-duration($age) || " days ago"
+                if ($age < xdt:dayTimeDuration("PT1H")) then
+                    let $minutes := if ($minutesSince = 1) then ' minute' else ' minutes'
+                        return $minutesSince || $minutes || " ago"
+                    
+                else if ($age < xdt:dayTimeDuration("P1D")) then
+                    let $hours := if ($hoursSince = 1) then ' hour' else ' hours'
+                        return $hoursSince || $hours || " ago"
+                    
+                else if ($age < xdt:dayTimeDuration("P14D")) then
+                    let $days := if ($daysSince = 1) then ' day' else ' days'
+                        return $daysSince || $days || " ago"
+                        
                     else
-                        format-dateTime($date, "[MNn] [D00], [Y0000]", "en", (), ())
+                        format-dateTime($date, "[MNn] [D00] [Y0000]")
```

Comment:
Please restore params 3-5, to ensure the date is displayed in English; otherwise, server locales will determine the language used.

Reply:
OK. Shall we put it in the timetunnel also, then? I copy-pasted the code from there, where those parameters weren't there in the first place.

Reply:
I think that would be outside the scope of this PR, but there's certainly an opportunity here to define a function (e.g., in the app module) where we generate dates.

Reply:
It sounds good, let's leave it as it is for the time being.